runtime/src/main/cpp/Memory.h

```diff
@@ -153,9 +165,17 @@ struct ContainerHeader {
   }
 
   inline void setColor(unsigned color) {
```

Comment:
As far as I understand, it makes sense to make the "default" coloring function ensuring that it doesn't recolor green object. I.e. to have `setColor` (asserting that current color is not green), `setColorUnlessGreen` and `setColorEvenIfGreen`. Btw, is the latter actually needed?